File: app/controller/JsonOutputStream.rb
require './StringOutputStream'
require 'json'

class JsonOutputStream
    def initialize output
        @stringOutputStream = StringOutputStream.new output
    end

    def write jsonData
        @stringOutputStream.write JSON.dump jsonData
    end
end

# test
# output = JsonOutputStream.new STDOUT
# jsonData = {
    # 'type' => 'log',
    # 'msg' => 'hello'
# }
# output.write jsonData

File: app/gui/JsonOutputStream.js
var StringOutputStream = require('./StringOutputStream');

var JsonOutputStream = function(output) {
    this.stringOutputStream = new StringOutputStream(output);
}

JsonOutputStream.prototype.write = function(jsonData) {
    var jsonStr = JSON.stringify(jsonData);
    this.stringOutputStream.write(jsonStr);
}

module.exports = JsonOutputStream;
